agent/statemanager/testdata/v23/perContainerRuntimeID/ecs_agent_data.json

```diff
@@ -0,0 +1,170 @@
+          "PullStoppedAt": "2018-10-04T18:05:48.445985904Z",
+          "ExecutionStoppedAt": "0001-01-01T00:00:00Z",
+          "SentStatus": "RUNNING",
+          "StartSequenceNumber": 2,
+          "StopSequenceNumber": 0,
+          "executionCredentialsID": "b1a6ede6-1a9f-4ab3-a02e-bd3e51b11244",
+          "ENI": null,
+          "MemoryCPULimitsEnabled": true,
+          "PlatformFields": {}
+        }
+      ],
+      "IdToContainer": {
+        "8f5e6e3091f221c876103289ddabcbcdeb64acd7ac7e2d0cf4da2be2be9d8956": {
+          "DockerId": "8f5e6e3091f221c876103289ddabcbcdeb64acd7ac7e2d0cf4da2be2be9d8956",
+          "DockerName": "ecs-private-registry-state-1-container1-a68ef4b6e0fba38d3500",
+          "Container": {
+            "Name": "container_1",
+            "Image": "amazonlinux:1",
+            "ImageID": "sha256:7f929d2604c7e504a568eac9a2523c1b9e9b15e1fcee4076e1411a552913d08e",
+            "Command": [
+              "sleep",
+              "3600"
+            ],
+            "Cpu": 0,
+            "Memory": 512,
+            "Links": null,
+            "volumesFrom": [],
+            "mountPoints": [],
+            "portMappings": [],
+            "Essential": true,
+            "EntryPoint": null,
+            "environment": {},
+            "overrides": {
+              "command": null
+            },
+            "dockerConfig": {
+              "config": "{}",
+              "hostConfig": "{\"CapAdd\":[],\"CapDrop\":[]}",
+              "version": "1.17"
+            },
+            "registryAuthentication": null,
+            "LogsAuthStrategy": "",
+            "desiredStatus": "RUNNING",
+            "KnownStatus": "RUNNING",
+            "RunDependencies": null,
+            "IsInternal": "NORMAL",
+            "ApplyingError": {
+              "error": "API error (500): Get https://registry-1.docker.io/v2/library/amazonlinux/manifests/1: toomanyrequests: too many failed login attempts for username or IP address\n",
+              "name": "CannotPullContainerError"
+            },
+            "SentStatus": "RUNNING",
+            "metadataFileUpdated": false,
+            "KnownExitCode": null,
+            "KnownPortBindings": null
+          }
+        }
+      },
+      "IdToTask": {
+        "8f5e6e3091f221c876103289ddabcbcdeb64acd7ac7e2d0cf4da2be2be9d8956": "arn:aws:ecs:us-west-2:1234567890:task/33425c99-5db7-45fb-8244-bc94d00661e4"
+      },
+      "ImageStates": [
+        {
+          "Image": {
+            "ImageID": "sha256:7f929d2604c7e504a568eac9a2523c1b9e9b15e1fcee4076e1411a552913d08e",
+            "Names": [
+              "amazonlinux:1"
+            ],
+            "Size": 165452304
+          },
+          "PulledAt": "2018-10-04T18:05:48.445644088Z",
+          "LastUsedAt": "2018-10-04T18:05:48.445645342Z",
+          "PullSucceeded": false
+        }
+      ],
+      "ENIAttachments": null,
+      "IPToTask": {}
+    }
+  },
+  "Version": 20
```

Comment:
this is not using state file version 23.. i think you need to build the agent with your current changes and run a task, then put the state file here